src/core/geometry/qgsbox3d.cpp

```diff
@@ -86,47 +112,110 @@ QgsBox3d QgsBox3d::intersect( const QgsBox3d &other ) const
 
 bool QgsBox3d::is2d() const
 {
-  return qgsDoubleNear( mZmin, mZmax ) || ( mZmin > mZmax );
+  return qgsDoubleNear( mZmin, mZmax ) || ( mZmin > mZmax ) || std::isnan( mZmin ) || std::isnan( mZmax ) ;
+}
+
+bool QgsBox3d::is3D() const
+{
+  return !is2d() && !isNull();
```

Comment:
Isn't possible to have a case where `( mZmin > mZmax )` and have to consider the box as an "inversed" 3D Box?

Reply:
I think, it's safe at the moment to assume that It returns `FALSE` if the box is not normalized. I have added a sentence about it in the function description.